edb/server/protocol/auth_ext/http.py

```diff
@@ -119,8 +117,90 @@ async def handle_request(
+                    if register_provider_id is None:
+                        raise errors.InvalidData(
+                            'Missing "provider" in register request'
+                        )
+
+                    local_client = local.Client(
+                        db=self.db, provider_id=register_provider_id
+                    )
+                    identity = await local_client.register(data)
+
+                    session_token = self._make_session_token(identity.id)
+                    response.custom_headers["Set-Cookie"] = (
+                        f"edgedb-session={session_token}; "
+                        f"HttpOnly; Secure; SameSite=Strict"
+                    )
+                    if data.get("redirect_to") is not None:
+                        response.status = http.HTTPStatus.FOUND
+                        response.custom_headers["Location"] = data[
+                            "redirect_to"
+                        ]
+                    else:
+                        response.status = http.HTTPStatus.CREATED
+
+                case ("authenticate",):
+                    content_type = request.content_type
+                    match content_type:
+                        case b"application/x-www-form-urlencoded":
+                            data = {
+                                k: v[0]
+                                for k, v in urllib.parse.parse_qs(
+                                    request.body.decode('ascii')
+                                ).items()
+                            }
+                        case b"application/json":
+                            data = json.loads(request.body)
+                        case _:
+                            raise errors.InvalidData(
+                                f"Unsupported Content-Type: {content_type}"
+                            )
+
+                    authenticate_provider_id = data.get("provider")
+                    if authenticate_provider_id is None:
+                        raise errors.InvalidData(
+                            'Missing "provider" in register request'
+                        )
+
+                    local_client = local.Client(
+                        db=self.db, provider_id=authenticate_provider_id
+                    )
+                    identity = await local_client.authenticate(data)
+
+                    session_token = self._make_session_token(identity.id)
+                    response.custom_headers["Set-Cookie"] = (
+                        f"edgedb-session={session_token}; "
+                        f"HttpOnly; Secure; SameSite=Strict"
+                    )
+                    if data.get("redirect_to") is not None:
+                        response.status = http.HTTPStatus.FOUND
+                        response.custom_headers["Location"] = data[
+                            "redirect_to"
+                        ]
+                    else:
+                        response.status = http.HTTPStatus.OK
```

Comment:
Feels like this should have a body, but I'm not exactly sure what to put in it just yet. Maybe the session claims?

Reply:
What would an app do/need here?

Reply:
The app needs to get the JWT into the globals to get the global `identity_id`. Currently we throw the session JWT in a cookie, and the server can parse it from there, but maybe we can provide it here?

Reply:
Sure, returning `identity_id` makes sense here

Reply:
I decided that it makes the most sense to actually have the full token in the payload here since the `identity_id` is computed rather than set, right?